tests/utils.go

```diff
@@ -1130,10 +1131,21 @@ func NewConsoleExpecter(virtCli kubecli.KubevirtClient, vmi *v1.VirtualMachineIn
 	expecterReader, expecterWriter := io.Pipe()
 	resCh := make(chan error)
 	go func() {
-		con, err := virtCli.VirtualMachineInstance(vmi.ObjectMeta.Namespace).SerialConsole(vmi.ObjectMeta.Name)
-		if err != nil {
-			resCh <- err
-			return
+		con, err := virtCli.VirtualMachineInstance(vmi.Namespace).SerialConsole(vmi.Name)
```

Comment:
I think the idea of making functional tests implicitly validate console waiting for scheduler is great and better than having one or two cases to cover this behavior explicitly.

But it doesn't seem that you really validate the change in the tool here. This code below is a duplicate of what is added in the tool; you don't reuse the code from there. I think the right thing to do is to reuse the code from the tool here. It may require spinning up some of the tool code into a separate helper function ready to use.

As-is, I suspect the tests changes would pass even without the fix in the tool.